pkg/endpointslice/controller_test.go

```diff
@@ -34,79 +30,79 @@ const testNS2 = "testNameSpace2"
 var _ = Describe("EndpointSlice controller", func() {
 	t := newEndpointSliceTestDiver()
 
-	When("Endpoints exists in an EndpointSlice ", func() {
-		When("IsHealthy is called for the remote clusters", func() {
-			It("should return the appropriate response", func() {
+	When("a service has a valid endpoint", func() {
+		When("IsHealthy is called for the service with a valid cluster", func() {
+			It("should return true", func() {
 				esName := testName1 + remoteClusterID1
 				endPoint1 := t.newEndpoint(cluster1HostNamePod1, cluster1EndPointIP1)
 				endpointSlice := t.newEndpointSliceFromEndpoint(testService1, remoteClusterID1, esName, testNS1, []v1beta1.Endpoint{endPoint1})
 				t.createEndpointSlice(testNS1, endpointSlice)
-				t.AwaitEndpointSlice(esName, testNS1)
-				t.AwaitEpMap(testService1, testNS1)
-				Expect(t.controller.IsHealthy(KeyFunc(testService1, testNS1), remoteClusterID1)).To(BeTrue())
+				t.awaitIsHealthy(testService1, testNS1, remoteClusterID1, false)
```

Comment:
I think it reads a little awkward to call `awaitIsHealthy` with false for the positive case.  I prefer separate functions: `awaitIsHealthy` for the positive case and `awaitNotIsHealthy` otherwise.